### PR TITLE
Feature/build framework refactor

### PR DESCRIPTION
**Pull request by adam-dg**

### CODE REVIEW

File: scripts/bootstrap-build-framework.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+set -e
+
+script_path=$(dirname $0)
+working_dir=$(pwd)
+cd "$script_path"
+
+if [ ! -d "./make" ]; then
+  version="master"
+  if [ -f ./.dbf ]; then
+    version=$(cat ./.dbf)
```
Comment (mikeddeeson) [inline scripts/bootstrap-build-framework.sh:12]: is this an optional file that can be defined against a project to set the version for future proofing the framework?

**Review by mikeddeeson — APPROVED** [review]

File: scripts/bootstrap-build-framework.sh (hunk 1 above)
Comment (adam-dg) [reply scripts/bootstrap-build-framework.sh:12]: yup

File: Makefile [hunk 2]
```diff
@@ -10,7 +10,11 @@ COMMAND=/bin/bash
 #
 
 # Shortcut for make install, make build and make start
-default: install build start
+default: bootstrap install build start
+
+# Bootstrap Deeson Drupal build framework
+bootstrap:
```
Comment (angry-dan) [inline Makefile:16]: I'm not sure that I see the difference between "bootstrap" and "install". I reckon just add this as the first command in the install target.

File: scripts/bootstrap-build-framework.sh (hunk 1 above)
Comment (angry-dan) [reply scripts/bootstrap-build-framework.sh:12]: Maybe the default version shouldn't be "master" but some tag? Doing it this way means you can never release a breaking change without having to immediately update all the other repos.

File: scripts/bootstrap-build-framework.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  fi
+
+  docker run -v $(pwd):/app -w /app --env DBFVER=$version deeson/deployer /bin/bash \
+    -c 'git clone -b "$DBFVER" --single-branch --depth 1 https://github.com/teamdeeson/drupal-build-framework.git make \
+    && cd ./make \
+    && rm -Rf .git'
```
Comment (angry-dan) [inline scripts/bootstrap-build-framework.sh:18]: I know everyone loves docker, but is there really any need to run a git operation inside of a container? Can we really not even assume that developer machines will have git installed?

**Pull request merged**